ext/node/ops/os/cpus.rs

```diff
@@ -0,0 +1,301 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::serde::Serialize;
+
+#[derive(Debug, Default, Serialize, Clone)]
+pub struct CpuTimes {
+  pub user: u64,
+  pub nice: u64,
+  pub sys: u64,
+  pub idle: u64,
+  pub irq: u64,
+}
+
+#[derive(Debug, Default, Serialize, Clone)]
+pub struct CpuInfo {
+  pub model: String,
+  pub speed: u64,
+  pub times: CpuTimes,
+}
+
+impl CpuInfo {
+  pub fn new() -> Self {
+    Self::default()
+  }
+}
+
+#[cfg(target_os = "macos")]
+pub fn cpu_info() -> Option<Vec<CpuInfo>> {
+  let mut model: [u8; 512] = [0; 512];
+  let mut size = std::mem::size_of_val(&model);
+
+  // Safety: Assumes correct behavior of platform-specific syscalls and data structures.
+  // Relies on specific sysctl names and sysconf parameter existence.
+  unsafe {
+    let ticks = libc::sysconf(libc::_SC_CLK_TCK);
+    let multiplier = 1000u64 / ticks as u64;
+    if libc::sysctlbyname(
+      "machdep.cpu.brand_string\0".as_ptr() as *const libc::c_char,
+      model.as_mut_ptr() as _,
+      &mut size,
+      std::ptr::null_mut(),
+      0,
+    ) != 0
+      && libc::sysctlbyname(
+        "hw.model\0".as_ptr() as *const libc::c_char,
+        model.as_mut_ptr() as _,
+        &mut size,
+        std::ptr::null_mut(),
+        0,
+      ) != 0
+    {
+      return None;
+    }
+
+    let mut cpu_speed: u64 = 0;
+    let mut cpu_speed_size = std::mem::size_of_val(&cpu_speed);
+
+    libc::sysctlbyname(
+      "hw.cpufrequency\0".as_ptr() as *const libc::c_char,
+      &mut cpu_speed as *mut _ as *mut libc::c_void,
+      &mut cpu_speed_size,
+      std::ptr::null_mut(),
+      0,
+    );
+
+    if cpu_speed == 0 {
+      // use a fixed value for frequency on arm64
+      cpu_speed = 2400000000;
+    }
```

Comment:
can you link to some documentation here? also use underscores in big values to clarify: `2_400_000_000`